compiler/rustc_hir_typeck/src/closure.rs

```diff
@@ -241,9 +242,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             if expected_sig.is_none()
                 && let ty::PredicateKind::Projection(proj_predicate) = bound_predicate.skip_binder()
             {
-                expected_sig = self.deduce_sig_from_projection(
+                expected_sig = self.normalize_associated_types_in(
```

Comment:
Fold after `deduce_sig_from_projection` so we only need to normalize the types that end up in the closure sig.